include/Core/Config.php

```diff
@@ -132,15 +132,33 @@ public static function set($family,$key,$value) {
 		$dbvalue = (is_array($value)?serialize($value):$value);
 		$dbvalue = (is_bool($dbvalue) ? intval($dbvalue) : $dbvalue);
 
-		$ret = q("INSERT INTO `config` ( `cat`, `k`, `v` ) VALUES ( '%s', '%s', '%s' )
-ON DUPLICATE KEY UPDATE `v` = '%s'",
+		// The "INSERT" command is very cost intense. It saves performance to do it this way.
```

Comment:
I would agree with your last sentence but in this case I feel like the readability took a big hit. And unless this function is called a lot during a single script execution, any performance improvement you will manage to implement won't improve the global performance significantly. That's why I suggested my counter-PR which specifically addresses the settings save pages where `set_config` and `set_pconfig` are called a lot in succession.

For the sake of argument, I ran the XDebug profiler against the current code, your branch and mine on the network page (the performance-critical page) after warmup.

Current code: 3 calls to `set_pconfig()` at ~46ms each = 137ms total, which is 3% of the total request time.
Your branch: 3 calls to `set_pconfig()` at ~1,2ms each = 3,5ms total, which is 0,28% of the total request time.
My branch: 3 calls to `set_pconfig()` at ~15ms each (45ms, 0 and 0) = 45ms total, which is 3% of the total request time.

Yours is undeniably faster, but that wasn't the point of this simplified benchmark. I'll just note that while the performance improvement on the function itself is staggering (4000% of improvement!), the overall impact on the performance critical page is rather low (2,5% shaved) and would probably be even lower on long-running maintenance tasks. My point rather was that I was able to get half the performance improvement you got with one line of cod while you added two SQL queries.

And while I salute your very successful effort at improving the performance of the `set_(p)config()` functions, I feel like the ratio readability / performance is less favorable with your code than mine.

The good news is that I found a way to recoup the difference between your result and mine. Friendica is constantly and uselessly profiling itself via the `App->save_timestamp()` that calls `debug_backtrace()` around 250 times in a single page. By turning off the profiling, it shaves around 45ms off the total loading time. I'll submit a separate PR for that.

In the mean time, I'm ready to beg you not to add those changes to Friendica, but ultimately the choice is yours.